### PR TITLE
fetch fixed to require XMLHttpRequest if not avaliable

### PR DESCRIPTION
**Pull request by enchev**
Fix for #1340

### CODE REVIEW

File: fetch/fetch.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ﻿(function () {
     'use strict';
 
+    if (!('XMLHttpRequest' in global)) {
+        var xhr = require("xhr");
+        global.XMLHttpRequest = xhr.XMLHttpRequest;
```
Comment (hdeshev) [inline fetch/fetch.js]: I think it's better if we create module-local `XMLHttpRequest` and `FormData` functions, use those, and avoid writing to the global object. This way we keep the "fetch" module clean of global initialization code, and we keep all our global init code in the "globals" module.

```javascript
var XMLHttpRequest = global.XMLHttpRequest;
if (!XMLHttpRequest) {
    var xhr = require("xhr");
    XMLHttpRequest = xhr.XMLHttpRequest
}
//The same for FormData
```

**Comment by enchev** [comment]
run ci

**Comment by hdeshev** [comment]
:+1:

**Comment by hshristov** [comment]
:+1:

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Pull request merged**